microprofile/fault-tolerance/src/test/java/io/helidon/microprofile/faulttolerance/AsynchronousBean.java

```diff
@@ -165,4 +172,47 @@ CompletableFuture<String> asyncCompletableFutureWithFallbackFailure() {
         future.completeExceptionally(new IOException("oops"));
         return future;
     }
+
+    /**
+     * Async call that uses a separate executor to run the method in a platform thread.
+     * A CDI producer must be present with the same {@code @WithExecutor} annotation.
+     *
+     * @return a future
+     * @see ExecutorServiceProducer#threadPoolSupplier()
+     */
+    @Asynchronous
+    @WithExecutor("platform-thread-executor")
+    CompletableFuture<String> asyncWithExecutor() {
+        Thread thread = Thread.currentThread();
+        called.set(thread.getName().contains("platform-threads") && !thread.isVirtual());
+        FaultToleranceTest.printStatus("AsynchronousBean::asyncWithExecutor", "success");
+        return CompletableFuture.completedFuture("success");
```

Comment:
Why not return the thread ref and doing the assertion in the test method? May be more clear what is wrong if the test fails

Reply:
Again, all these tests can be improved, but that's for another PR.